### PR TITLE
HEEDLS-411 - implement StoreASPProgressNoSession tracker endpoint

### PR DESCRIPTION
**Pull request by DanBloxham-sw**
### JIRA link
https://softwiretech.atlassian.net/browse/HEEDLS-411

### Description
Implemented the new StoreASPProgressNoSession tracker endpoint. As a lot of the logic is shared with the StoreASPProgressV2 endpoint, a good chunk of these changes are just moving that endpoint's methods and tests into a reusable format.
Note that there are still a few questions regarding differences in ticket scope vs the old code that are awaiting answers from Kevin when he gets back.

### Screenshots
N/A

-----
### Developer checks
(Leave tasks unticked if they haven't been appropriate for your ticket.)

I have:
- [x] Run the formatter and made sure there are no IDE errors.
- [x] Written tests for the changes (accessibility tests, unit tests for controller, data services, services, view models, etc)
- [x] Manually tested my work with and without JavaScript. Full manual testing guidelines can be found here: https://softwiretech.atlassian.net/wiki/spaces/HEE/pages/6703648740/Testing
- [ ] Updated/added documentation in Swiki and/or Readme. Links (if any) are below:
  - [doc_1_here](link_1_here)
- [x] Updated my Jira ticket with information about other parts of the system that were touched as part of the MR and have to be sanity tested to ensure nothing’s broken.
- [x] Scanned over my own MR to ensure everything is as expected.

### CODE REVIEW

File: DigitalLearningSolutions.Data/DataServices/SessionDataService.cs [hunk 1]
```diff
@@ -73,5 +78,28 @@ public bool HasAdminGotSessions(int adminId)
...
+                    FROM Sessions WHERE SessionID = @sessionId",
+                new { sessionId }
+            );
+        }
+
+        public void AddTutorialTimeToSessionDuration(int sessionId, int tutorialTime)
+        {
+            connection.Query(
+                @"UPDATE Sessions SET Duration = Duration + @tutorialTime
+                   WHERE [SessionID] = @sessionId",
```
Comment (AlexJacksonDS) [inline DigitalLearningSolutions.Data/DataServices/SessionDataService.cs:100]: Do we need the square brackets?

**Review by AlexJacksonDS — COMMENTED** [review]
Looks fine, just a minor comment on SQL formatting

File: DigitalLearningSolutions.Data/Services/StoreAspProgressService.cs [hunk 2]
```diff
@@ -0,0 +1,128 @@
+﻿namespace DigitalLearningSolutions.Data.Services
+{
+    using DigitalLearningSolutions.Data.DataServices;
+    using DigitalLearningSolutions.Data.Enums;
+    using DigitalLearningSolutions.Data.Models.Progress;
+
+    public interface IStoreAspProgressService
+    {
+        public (TrackerEndpointResponse? validationResponse, DetailedCourseProgress? progress)
```
Comment (livzorn) [inline DigitalLearningSolutions.Data/Services/StoreAspProgressService.cs:9]: remove the public keyword here?

**Review by livzorn — COMMENTED** [review]
Nice refactoring! Only spotted one small thing

**Review by livzorn — APPROVED** [review]

**Pull request merged**